src/ensembl/src/content/app/entity-viewer/state/gene-view/entityViewerGeneViewReducer.ts

```diff
@@ -45,7 +51,45 @@ export default function entityViewerGeneViewReducer(
           [activeObjectId]: updatedStateFragment
         }
       };
+    case getType(proteinsActions.updateGeneViewProteinsUIState):
+      return {
+        ...state,
+        [action.payload.activeGenomeId]: {
+          ...state[action.payload.activeGenomeId],
+          [action.payload.activeObjectId]: {
+            ...state[action.payload.activeGenomeId][
+              action.payload.activeObjectId
+            ],
+            contentUI: {
+              ...state[action.payload.activeGenomeId][
+                action.payload.activeObjectId
+              ].contentUI,
+              protein: proteinsReducer(state, action)
```

Comment:
No, what I meant was we would split gene view reducer into smaller reducers and combine them with combineReducers, as in `entityViewerReducer`

Reply:
I tried that to use `combineReducers` but I had no luck getting it to work without changing the current structure of the state.

Reply:
Ok, as discussed at the code review, we will try to move to `redux-toolkit` anyway, which will make us reconsider this piece of state later. It's fine if it works for now.